dev_setup.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/sh
+
+# cryptol setup script
+#
+# Supported distribution(s):
+#  * macOS 14 (AArch64)
+#
+# This script installs everything needed to get a working development
+# environment for cryptol. Any new environment requirements should be
+# added to this script.
+#
+
+set -e
+
+HERE=$(cd `dirname $0`; pwd)
+LOG=$HERE/dev_setup.log
+ROOT=$HERE
+ENV=$ROOT/env.sh
+
+WHAT4_SOLVERS_SNAPSHOT="snapshot-20240212"
+WHAT4_SOLVERS_URL="https://github.com/GaloisInc/what4-solvers/releases/download/$WHAT4_SOLVERS_SNAPSHOT/"
+WHAT4_SOLVERS_MACOS_12="macos-12-X64-bin.zip"
+WHAT4_SOLVERS_MACOS_14="macos-14-ARM64-bin.zip"
+
+function notice {
+    echo "[NOTICE] $*"
+}
+
+# Requires: LOG set to log file path.
+function logged {
+    if ! [ -z "$LOG" ]
+    then
+        mkdir -p `dirname $LOG`
+        echo "$@" >>$LOG
+        if ! "$@" >>$LOG 2>&1
+        then
+            echo
+            echo "An error occurred; please see $LOG"
+            echo "Here are the last 50 lines:"
+            tail -n 50 $LOG
+            exit 1
+        fi
+    fi
+}
+
+function update_submodules {
+    cd $ROOT
+    notice "Updating submodules"
+    logged git submodule update --init
+}
+
+function install_ghcup {
+    if ! ghcup --version &> /dev/null
+    then
+        notice "Installing ghcup, GHC, and cabal"
+        # Technically the installation only requires cabal, but it's
+        # recommended to get the whole GCH shebang in one package.
+        # The output is not routed to log because the installer is interactive.
+        curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
+    else
+        notice "Using existing ghcup installation"
+    fi
+
+}
+
+function is_macos {
+    [ "$OSTYPE" = "darwin"* ]
```

Comment:
The * is a bashism. Also I don't think $OSTYPE is standard; it's not set in sh on a random Ubuntu VM I just checked. Probably what you want is to use ```$(uname -s)``` and test it for "Darwin" because that way you don't need to mask off a version number, plus uname is portable. (Or portable enough. It's missing on some things like AIX that we definitely don't care about.)

Reply:
Oh, thanks. I used `uname` instead. I think there are some failure modes where e.g. someone is running macos 10 which has the same category and hardware as macos 12 but I'm (a) not sure how to detect exactly the right thing we need and (b) don't expect anyone is going to try to run this on older macOS versions or, like, an iPad.

Reply:
```uname -r``` prints the OS version (unfortunately on macOS it's not the MacOS version but the internal version of the kernel, or something, anyway it's 22.* for Ventura) and ```uname -m``` and ```uname -p``` print different variations of the machine type. Those are all portable (except e.g. AIX); there's also probably an apple-specific way to get "13" for Ventura or even the string "Ventura" but I don't know offhand what it is.

anyway info is available -- I suspect we don't actually need to care about the OS version though, just whether it's x86 or arm.

Reply:
(I guess we do need the OS version as well as the platform to get the right solver bins)

Reply:
This is now superseded by the `supported_platform` method which I think doesn't use bashisms or un-portable methods!

Reply:
I'll go look at the new stuff in a bit :-)